pkg/operator/status_test.go

```diff
@@ -93,3 +93,62 @@ func TestApplyConditions(t *testing.T) {
 	// will be off.
 	assert.Equal(t, co_check.Status.Conditions[0].Status, configv1.ConditionTrue, "expected same conditions")
 }
+
+func TestCompareVersions(t *testing.T) {
+	type tCase struct {
+		oV           []osconfigv1.OperandVersion
+		expectedBool bool
+		expectedErr  error
+	}
+	tCases := []tCase{
+		{
+			oV: []osconfigv1.OperandVersion{
+				{
+					Name:    "operator",
+					Version: "1.0",
+				},
+			},
+			expectedBool: true,
+			expectedErr:  nil,
+		},
+		{
+			oV: []osconfigv1.OperandVersion{
+				{
+					Name:    "operator",
+					Version: "2.0",
+				},
+			},
+			expectedBool: false,
+			expectedErr:  nil,
+		},
+	}
+	desiredVersions := []osconfigv1.OperandVersion{
+		{
+			Name:    "operator",
+			Version: "2.0",
+		},
+	}
+	co := &osconfigv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: "cluster-autoscaler"}}
+	for i, tc := range tCases {
+		co.Status.Versions = tc.oV
+		r := StatusReporter{
+			client:         fakeconfigclientset.NewSimpleClientset(co),
+			relatedObjects: []configv1.ObjectReference{},
+		}
+		isProgress, err := r.IsDifferentVersions(desiredVersions)
+		assert.Equal(t, tc.expectedBool, isProgress, "case %v: return expected %v but didn't get it", i, tc.expectedBool)
+		assert.Equal(t, tc.expectedErr, err, "case %v: expected %v error but didn't get it, got: ", i, tc.expectedErr, err)
+	}
+}
+
+func TestPrintOperandVersions(t *testing.T) {
+	desiredVersions := []osconfigv1.OperandVersion{
+		{
+			Name:    "operator",
+			Version: "2.0",
+		},
+	}
+	expectedString := "operator: 2.0"
```

Comment:
Given that `[print]operandVersions()` joins the `desiredVersions` together does it make sense to expand the test cases here to cover that case?

Reply:
Good idea.